### PR TITLE
Na new redux

### PR DESCRIPTION
**Pull request by nitya308**
# Title

Redux and screens! Please take a look at auth specifically?

### CODE REVIEW

File: packages/app/actions/forestActions.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  return (dispatch) => {
+    axios
+      .post<ForestCreateResponse>(BASE_URL, createForestRequest)
+      .then((response) => {
+        // RESPONSE HAS THE BELOW DATATYPE
+        // forest: Forest;
+        dispatch({
+          type: ForestActionTypes.SYNC_ALL_FORESTS,
+          payload: [response.data.forest], //this is converted to an array because this reducer expects an array of forests
+        });
```
Comment (mirunaish) [inline packages/app/actions/forestActions.ts:28]: using sync for this is probably not a good idea. i would make a new action for adding a new forest. see my comment on the forest reducer

File: packages/app/actions/index.ts [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+export const ActionTypes = {
+  SET_INTERNAL_ERROR: "SET_ERROR",
+  CLEAR_INTERNAL_ERROR: "CLEAR_ERROR",
+  SET_SYNC_ERROR: "SET_SYNC_ERROR",
+  CLEAR_SYNC_ERROR: "CLEAR_SYNC_ERROR",
+  SYNC_PROJECT_GENERATIONS: "SYNC_PROJECT_GENERATIONS",
+  SYNC_SPECIES: "SYNC_SPECIES",
+  SYNC_PLOT_FIELD_VALUES: "SYNC_PLOT_FIELD_VALUES",
+  SYNC_FIELDS: "SYNC_FIELDS",
+  CREATE_EDIT_FIELDS: "CREATE_FIELD",
```
Comment (mirunaish) [inline packages/app/actions/index.ts:15]: not important but probably want to change the value here to match the key

File: packages/app/actions/projectActions.ts [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+          payload: response.data.project,
+        });
+        // newfields: FieldCreateResponse[];
+        dispatch({
+          type: ActionTypes.CREATE_EDIT_FIELDS,
+          payload: response.data.newfields.map(function (
+            f: FieldCreateResponse
+          ) {
+            return f.field;
+          }),
```
Comment (mirunaish) [inline packages/app/actions/projectActions.ts:66]: i'm pretty sure editing a project returns a list of all fields, not just new or edited ones. so you'd have to replace all the fields in the local data with the new fields returned by the backend

Comment (mirunaish) [reply packages/app/actions/projectActions.ts:66]: i think i changed this response type in the backend a while ago

**Review by mirunaish — COMMENTED** [review]
auth stuff looks good

File: packages/app/reducers/forestReducer.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+const ForestReducer = produce((draftState: ForestState, action = {}) => {
+  draftState = initialState;
+  switch (action.type) {
+    case ForestActionTypes.SELECT_FOREST:
+      draftState.selected = action.payload;
+      break;
+    case ForestActionTypes.SYNC_ALL_FORESTS:
+      const newForests: [Forest] = action.payload;
+      newForests.forEach((newForest) => {
+        draftState.all[newForest.id] = newForest;
```
[the remark this answers is not in the document]
Comment (nitya308) [reply packages/app/reducers/forestReducer.ts]: working on local storage stuff soon!

File: packages/app/actions/forestActions.ts (hunk 1 above)
Comment (nitya308) [reply packages/app/actions/forestActions.ts:29]: done! I'm going to update all other syncs to add too!

File: packages/app/components/ForestDropdownMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+// only used for testing purposes
+const forestsData = [
+  { name: "Forest 1", id: 1 },
+  { name: "Forest 2", id: 2 },
+  { name: "Forest 3", id: 3 },
+  { name: "Forest 4", id: 4 },
+];
+
+export const ForestDropdownMenu = () => {
+  const [selectedValue, setSelectedValue] = useState("forest");
```
[the remark this answers is not in the document]
Comment (nitya308) [reply packages/app/components/ForestDropdownMenu.tsx]: I think this was a temp thing done by Marvin? I changed this completely!

**Pull request merged**